task/upload.go

```diff
@@ -539,23 +549,26 @@ func uploadTaskOutputLocations(tctx *TaskContext) ([]OutputName, []clients.Outpu
 	return outputNames, outputLocations, nil
 }
 
-func outputLocations(outURL string, relativePath string, autoMp4s bool) ([]OutputName, []clients.OutputLocation, error) {
+func outputLocations(outURL string, hls string, hlsRelPath string, mp4 string, mp4RelPath string) ([]OutputName, []clients.OutputLocation, error) {
 	url, err := url.Parse(outURL)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error parsing object store URL: %w", err)
 	}
-	// source segments cannot be stored in web3.storage
-	sourceSegments := !strings.HasPrefix(outURL, "w3s")
 	names, locations :=
 		[]OutputName{OutputNameOSPlaylistHLS},
```

Comment:
You have to add another value to this slice here. Let me share a bit more context of why this `outputLocations` function exists (my bad for not having documented it originally, this was all done in a rush): 
 - The original Catalyst API provided no way to associated outputs from the requested output_locations
 - It actually ended up supporting only 1 location and 1 output in the first version, but it received and returned an array
 - There would also be other output types that would be incompatible with sending everything on a single output_location: e.g. saving to IPFS, which would have a completely different URL with pinata creds instead of S3

So we created this function here, which would create outputs in a deterministic order and then we did a logic on `processCatalystCallback` that assumes Catalyst also returns the `outputs` on the exact order as they were requested [1]. That's the whole reason for having those `names` here which are also returned and then processed in order [on the big switch in the process result function](https://github.com/livepeer/task-runner/blob/b112d7458c808c7d01f54afdab2f06ccae706368/task/upload.go#L332).

[1] (but we don't actually use the same request data from the task params, we regenerate the outputs which is error prone in case of deploys)

I'm not sure if either of those constraints are still true. Maybe we can already associate specific `output_locations` with their corresponding `outputs` now. Or maybe the output data became different enough that we can actually tell which is which without trying to associate.

Either way, I never liked this logic, so feel free to refactor it completely. For now though, since you are adding a new output location, you might need to add a new output name and make sure that `processCatalystCallback` is able to process it consistently. In the long run it'd be good to have a safer way to associate the 2 things though, like an output_location id, or maybe the outputs just embed the entire output_location requests, idk!

Reply:
Ok, I get it now. It's very confusing, for now, I'll add another output here, but we should think about tidying it up a little bit.

Reply:
Yeah agreed. We need some reliable way to map outputs to the requests somehow